### PR TITLE
Disable Page Template picker if modal layout picker is available

### LINKED ISSUE #2419: [Modal Layout Picker] Create Modal window for Modal Layout Picker
### Is your feature request related to a problem? Please describe.
This task is part of the effort to rework the Starter Page Template Picker for new pages. This part of the initiative will focus on creating the base modal for the new Modal Layout Picker. Other tasks will then work off of the modal created in this task. It will also handle the feature flags to set up the build environments to enable the development and building of the new modal.

The goal will be to create a UI that looks something like:

| iOS | Android |
| --- | --- | 
| iOS-Container | Android-Container |

### Goals - Modal Container
+ When creating a new page, a modal should display with the following items:
    + Title
    + Subtitle
    + "Create Blank Page" Button
    + A button for dismissing the modal. 
    + These items should be styled for Light and Dark interfaces. 
+ Selecting the "Create Blank Page" button should create a new blank page.
+ Selecting the "X" (iOS) or Up button (Android) should dismiss the modal.
    + Android should validate the back button works as expected for dismissing the modal.
+ Swiping down on the modal should dismiss the modal. 

### Goals - Feature Flag
+ Selecting the Create page should only show this new modal when running:
    + iOS build configurations: `localDevelopment` and `a8cBranchTest`.
    + Android build configurations: `vanilla` and `jalapeno`.
+ Opening the editor **should not** show the existing STP picker if the modal **has** been shown
    + We can pass the feature flag to RN as part of capabilities ([iOS](https://github.com/WordPress/gutenberg/blob/7532a485b400f86638145b71f94f6f717e5add25/packages/react-native-bridge/ios/GutenbergBridgeDataSource.swift#L43), [Android](https://github.com/WordPress/gutenberg/blob/7532a485b400f86638145b71f94f6f717e5add25/packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java#L470))
    + The key to pass as part of the capabilities should be `modalLayoutPicker`. `true` would mean the app supports modal layout picker
+ Opening the editor **should** show the existing STP picker if the modal **has not** been shown

### Additional context
+ Creating a new page can happen from:
    + The FAB button on the bottom right of the site's home screen
    + The ➕ on the top right on the "Pages" screen in iOS or the FAB button on the "Pages" screen in Android

<details>
<summary>Blueprint Flow - iOS</summary>

</details>

<details>
<summary>Blueprint Flow - Android</summary>

</details>

### PR DESCRIPTION
**Pull request by chipsnyder**
Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/2419

Related PRs:
- `gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2478
- `WordPress-iOS`: https://github.com/wordpress-mobile/WordPress-iOS/pull/14456

## Description
This change enables the mobile editor to accept the capability `modalLayoutPicker` and based on that conditionally show or hide the Template Picker.

Android hasn't started Modal Layout Picker yet so this only needs to 

## How has this been tested?
To disable or enable the development version of Modal Layout Picker
- Open the app from the build https://github.com/wordpress-mobile/WordPress-iOS/pull/14456#issuecomment-656759183 or a local development build
    - By default, the modal layout picker will be enabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Gutenberg Modal Layout Picker" to enable or disable the picker
            <details>
            <summary>To toggle: </summary>
            
            </details>

### On iOS using the build options mentioned above
- With Gutenberg Modal Layout Picker toggled on:
    - Select to create a new page
    - Expect to see the container for the new modal Layout Picker
    - Select "Create Blank Page"
    - Expect the current template picker to be hidden

- With Gutenberg Modal Layout Picker toggled on:
    - Select an existing page or draft without content (just a title)
    - You should not see the new Modal Layout Picker
    - Expect the current template picker to be hidden

- With Gutenberg Modal Layout Picker toggled off:
    - Select to create a new page
    - You should not see the new Modal Layout Picker
    - Expect the current template picker to be shown. 

- With Gutenberg Modal Layout Picker toggled off:
    - Select an existing page or draft without content (just a title)
    - You should not see the new Modal Layout Picker
    - Expect the current template picker to be shown

- With Gutenberg Modal Layout Picker toggled on or off:
    - Select an existing page or draft with content
    - Expect the current template picker to be hidden

### On Android running a metro server or generating a build

- Existing Blank Page
    - Select an existing page or draft without content (just a title)
    - Expect the current template picker to be shown
- New Page
    - Select to create a new page
    - Expect the current template picker to be shown. 
- Existing Page
    - Select an existing page or draft with content
    - Expect the current template picker to be hidden

## Types of changes
New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by chipsnyder** [comment]
The iOS PR is still in design review but this can be reviewed and merged separately from that.

File: packages/block-editor/src/components/page-template-picker/with-page-template-picker.native.js [hunk 1]
```diff
@@ -13,8 +13,15 @@ import {
...
 const __experimentalWithPageTemplatePicker = createHigherOrderComponent(
 	( WrappedComponent ) => ( props ) => {
-		const isTemplatePickerVisible = __experimentalUsePageTemplatePickerVisible();
-		const isTemplatePickerAvailable = __experimentalUsePageTemplatePickerAvailable();
+		const isTemplatePickerEnabled = ! (
+			props.capabilities.modalLayoutPicker ?? false
+		);
+		const isTemplatePickerVisible =
+			__experimentalUsePageTemplatePickerVisible() &&
+			isTemplatePickerEnabled;
```
Comment (geriux) [inline packages/block-editor/src/components/page-template-picker/with-page-template-picker.native.js:21]: What do you think if we switch the order to:

```suggestion
			isTemplatePickerEnabled &&
			__experimentalUsePageTemplatePickerVisible();
```

This way we avoid executing `__experimentalUsePageTemplatePickerVisible` if `isTemplatePickerEnabled` is false.

File: packages/block-editor/src/components/page-template-picker/with-page-template-picker.native.js [hunk 2]
```diff
@@ -13,8 +13,15 @@ import {
...
+		const isTemplatePickerAvailable =
+			__experimentalUsePageTemplatePickerAvailable() &&
```
Comment (geriux) [inline packages/block-editor/src/components/page-template-picker/with-page-template-picker.native.js:23]: Same as the comment above =)

**Comment by geriux** [comment]
Left some suggestions but it's working well with the test cases provided 🎉

**Comment by chipsnyder** [comment]
Thanks for the suggestion @geriux! This has been updated now

**Review by geriux — APPROVED** [review]
LGTM! Tested on iOS and Android.

**Pull request merged**